homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,268 @@
+        _LOGGER.error('Could not add the FSAPI device at %s:%s -> %s',
+                      host, port, password)
+
+    return False
+
+
+class FSAPIDevice(MediaPlayerDevice):
+    """Representation of a Frontier Silicon device on the network."""
+
+    def __init__(self, device_url, password):
+        """Initialize the Frontier Silicon API device."""
+        self._device_url = device_url
+        self._password = password
+        self._state = STATE_UNKNOWN
+
+        self._name = None
+        self._title = None
+        self._artist = None
+        self._album_name = None
+        self._mute = None
+        self._source = None
+        self._source_list = None
+        self._media_image_url = None
+
+    def get_fs(self):
+        """Create a fsapi session."""
+        from fsapi import FSAPI
+
+        return FSAPI(self._device_url, self._password)
+
+    # Properties
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return True
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._name
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._title
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._album_name
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_FRONTIER_SILICON
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        return self._state
+
+    # source
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._source
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._media_image_url
+
+    def update(self):
+        """Get the latest date and update device state."""
+        fs_device = self.get_fs()
+
+        if not self._name:
+            self._name = fs_device.friendly_name
+
+        if not self._source_list:
+            self._source_list = fs_device.mode_list
+
+        info_name = fs_device.play_info_name
+        info_text = fs_device.play_info_text
+
+        title = ''
+        if info_name:
+            if title:
+                title += ' - '
+            title += info_name
+        if info_text:
+            title += ': ' + info_text
+
+        self._title = title
+
+        self._artist = ''
```

Comment:
This can just be `self._artist = fs_device.play_info_artist`. Hass will treat media fields that return `None` as being not specified. Same for album_name below.